frontend/resourceadm/components/NewPartyListModal/NewPartyListModal.test.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useRef } from 'react';
+import { MemoryRouter } from 'react-router-dom';
+import { render as rtlRender, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { act } from 'react-dom/test-utils';
+import { textMock } from '../../../testing/mocks/i18nMock';
+import { queriesMock } from 'app-shared/mocks/queriesMock';
+import { NewPartyListModal } from './NewPartyListModal';
+import { ServicesContextProps, ServicesContextProvider } from 'app-shared/contexts/ServicesContext';
+import { createQueryClientMock } from 'app-shared/mocks/queryClientMock';
+
+const mockButtonText: string = 'Mock Button';
+const closeModalMock = jest.fn();
+
+const defaultProps = {
+  org: 'orgname',
+  env: 'tt02',
+  navigateUrl: '/listadmin/partylist/',
+  onClose: closeModalMock,
+};
+
+const mockedNavigate = jest.fn();
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useNavigate: () => mockedNavigate,
+}));
+
+const user = userEvent.setup();
+
+describe('NewPartyListModal', () => {
+  it('should disable create button when name or id is empty', async () => {
+    await renderAndOpenModal();
+
+    const createButton = screen.getByText(textMock('resourceadm.listadmin_confirm_create_list'));
+    expect(createButton).toHaveAttribute('aria-disabled', 'true');
+  });
+
+  it('should close modal on cancel click', async () => {
+    await renderAndOpenModal();
+
+    const closeButton = screen.getByText(textMock('general.cancel'));
+    await act(() => user.click(closeButton));
+
+    expect(closeModalMock).toHaveBeenCalled();
+  });
+
+  it('should navigate after party list is created', async () => {
+    await renderAndOpenModal();
+
+    const nameField = screen.getByLabelText(textMock('resourceadm.listadmin_list_name'));
+    await act(() => user.type(nameField, 'nytt navn'));
+
+    const createButton = screen.getByText(textMock('resourceadm.listadmin_confirm_create_list'));
+    await act(() => user.click(createButton));
+
+    await waitFor(() => {
+      expect(mockedNavigate).toHaveBeenCalledWith('/listadmin/partylist/nytt-navn');
+    });
+  });
+
+  it('should show error message when trying to create a party list with an existing identifier', async () => {
+    await renderAndOpenModal({
+      createPartyList: jest
+        .fn()
+        .mockImplementation(() => Promise.reject({ response: { status: 409 } })),
```

Comment:
Could we maybe use the ```ServerCode``` enum instead of the number code for the status? 
```suggestion
        .mockImplementation(() => Promise.reject({ response: { status: ServerCodes.Conflict } })),
```